climada/hazard/storm_europe.py

```diff
@@ -199,6 +206,209 @@ def _read_one_nc(self, file_name, centroids):
         ncdf.close()
         return new_haz
 
+
+    def read_cosmoe_file(self, fp_file, run_date, event_date=None,
+                         model_name='COSMO-2E', description=None):
+        """Clear instance and read MeteoSwiss cosmoe weather forecast
+        footprint netcdf files into it. One event is one full day in UCT.
+        Works for COSMO-1E and COSMO-2E
+
+        Parameters:
+            fp_file (str): string directing to one netcdf file
+            run_date (datetime): The starting timepoint of the forecast run
+                of the cosmo model
+            event_date (datetime, optional): one day within the forecast
+                period, only this day (00H-24H) will be included in the hazard
+            model_name (str,optional): select the name of the icon model to
+                be downloaded. Must match the url string
+                (see _download_icon_grib for further info)
+            description (str, optional): description of the events, defaults
+                to 'cosmo weather forecast'
+        """
+        self.clear()
+        # create centroids
+        self.centroids = self._centroids_from_nc(fp_file)
+
+        # read intensity from file
+        ncdf = xr.open_dataset(fp_file)
+        ncdf = ncdf.assign_coords(date=('time',ncdf["time"].dt.floor("D")))
+
+        if event_date:
+            # stacked = np.max(ncdf.sel(time=event_date.strftime('%Y-%m-%d')).VMAX_10M,
+            #                  axis=0).stack(intensity=('y_1', 'x_1'))
+            stacked2 = ncdf.sel(time=event_date.strftime('%Y-%m-%d')
+                                ).groupby('date'
+                                          ).max().stack(intensity=('y_1',
+                                                                   'x_1'))
+            considered_dates = np.datetime64(event_date)
+        else:
+            time_covered_step = ncdf['time'].diff('time')
+            time_covered_day = time_covered_step.groupby('date').sum()
+            # forecast run should cover at least 18 hours of a day
+            days_to_consider = time_covered_day > np.timedelta64(18,'h') 
+            stacked2 = ncdf.groupby('date'
+                                    ).max().sel(date=days_to_consider
+                                                ).stack(intensity=('y_1',
+                                                                   'x_1'))
+            unique_dates = ncdf['time'].groupby('date').min()
+            considered_dates = unique_dates.sel(date=days_to_consider).values
```

Comment:
necessary to create another var `unique_dates` here? or could this be directly extracted from `stacked2`

Reply:
Yes, implemented